src/test/java/com/redhat/cloud/notifications/GwResourceTest.java

```diff
@@ -2,45 +2,67 @@
 
 import io.quarkus.test.common.QuarkusTestResource;
 import io.quarkus.test.junit.QuarkusTest;
-import org.junit.jupiter.api.Disabled;
+import io.vertx.core.json.Json;
 import org.junit.jupiter.api.Test;
 
 import javax.ws.rs.core.MediaType;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
 
 import static io.restassured.RestAssured.given;
-import static org.hamcrest.CoreMatchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
 
 @QuarkusTest
 @QuarkusTestResource(TestLifecycleManager.class)
 public class GwResourceTest {
 
-    @Disabled("Class loading issues in Quarkus testing apparently")
     @Test
-    public void testHelloEndpoint() {
+    public void testHelloEndpoint() throws InterruptedException {
+
+        UUID random = UUID.randomUUID();
 
         RestAction ra = new RestAction();
-        ra.accountId="42";
-        ra.bundle="bundle-test";
-        ra.application="test";
-        ra.eventType="hulla";
+        ra.accountId = "42";
+        ra.bundle = "bundle-test";
+        ra.application = "test";
+        ra.eventType = "hulla";
         ra.payload = new HashMap();
-        ra.payload.put("key1","value1");
+        ra.payload.put("key1", "value1");
+        ra.payload.put("uuid",random);
+
         DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME;
-        ra.timestamp=LocalDateTime.now().format(formatter);
+        ra.timestamp = LocalDateTime.now().format(formatter);
         ra.setTimestamp("2020-12-11T12:13:03.507753");
 
-        String identity = TestHelpers.encodeIdentityInfo("test","user");
+        String identity = TestHelpers.encodeIdentityInfo("test", "user");
 
         given()
-            .body(ra)
-            .header("x-rh-identity",identity)
-            .contentType(MediaType.APPLICATION_JSON)
-          .when().post("/notifications/")
-          .then()
-             .statusCode(200);
-    }
+                .body(ra)
+                .header("x-rh-identity", identity)
+                .contentType(MediaType.APPLICATION_JSON)
+                .when().post("/notifications/")
+                .then()
+                .statusCode(200);
 
+
+        // Now check if we got a message
+        for (int i = 0; i < 10; i++) {
+            Thread.sleep(1000L);
```

Comment:
I hope that this doesn't mean that we will have to restart this test multiple times for it to pass :)

Reply:
Normally it triggers in the 1st round. I just put the loop in for slower build systems, where it may take longer, so that the test does not abort if it takes a bit more than a second. With the loop we don't need to wait a fixed amount of say 10s or so.